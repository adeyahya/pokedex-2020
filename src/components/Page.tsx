import { FunctionComponent, ReactNode, memo } from 'react';
import { useRouter } from 'next/dist/client/router';
import Link from 'next/link';

interface Props {
  children: ReactNode;
}

export const Page: FunctionComponent<Props> = memo(({ children }) => {
  const router = useRouter();
  const isHomepage = router.pathname === '/';

  return (
    <div>
      <nav className="flex items-center justify-between flex-wrap bg-indigo-900 p-5">
        <div className="flex items-center flex-shrink-0 text-white mr-6">
          {!isHomepage && (
            <button onClick={router.back} className="lg:hidden mr-4">
              <svg
                viewBox="64 64 896 896"
                focusable="false"
                data-icon="arrow-left"
                width="1.4rem"
                height="1.4rem"
                fill="currentColor"
                aria-hidden="true"
              >
                <path d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"></path>
              </svg>
            </button>
          )}

          <Link href="/">
            <a>
              <span className="font-semibold text-xl tracking-tight">Pokedex</span>
            </a>
          </Link>
        </div>
      </nav>
      {children}
    </div>
  );
});
