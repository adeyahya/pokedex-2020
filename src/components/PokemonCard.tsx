import { FunctionComponent, useCallback } from 'react';
import Link from 'next/link';
import { PokemonFragment } from '../queries/types/PokemonFragment';
import { SpecsCard } from './SpecsCard';
import { SpecListCard } from './SpecListCard';
import { useRouter } from 'next/dist/client/router';

interface Props {
  pokemon: PokemonFragment | null;
  prefixUrl?: string;
}

export const PokemonCard: FunctionComponent<Props> = ({ pokemon, prefixUrl = '/' }) => {
  const router = useRouter();
  const targetUrl = prefixUrl.concat(pokemon?.id || '');

  const handleClick = useCallback(() => {
    router.push(targetUrl);
  }, [targetUrl]);

  return (
    <div
      onClick={handleClick}
      className="shadow hover:shadow-lg bg-white rounded-lg cursor-pointer"
      data-testid={`pokemon-card-${pokemon?.name}`}
    >
      <figure className="h-20 md:h-32 lg:h-40 flex items-center justify-center pt-4">
        <img
          className="h-full"
          src={pokemon?.image || ''}
          alt={pokemon?.name || ''}
          data-testid="image"
        />
      </figure>
      <div className="p-4">
        <Link href={targetUrl}>
          <a>
            <h3 className="text-lg font-bold mb-0" data-testid="name">
              {pokemon?.name}
            </h3>
          </a>
        </Link>
        <span className="text-sm italic mb-2 block" data-testid="classification">
          {pokemon?.classification}
        </span>
        <div className="mb-3">
          <SpecListCard list={pokemon?.types || []} label="types" />
        </div>
        <div className="mt-2">
          <SpecsCard
            specs={{
              minWg: pokemon?.weight?.minimum,
              maxWg: pokemon?.weight?.maximum,
              maxHP: pokemon?.maxHP,
              maxCP: pokemon?.maxCP,
            }}
          />
        </div>
      </div>
    </div>
  );
};
