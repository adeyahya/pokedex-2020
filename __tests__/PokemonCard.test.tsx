import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import { PokemonCard } from '../src/components/PokemonCard';
import TEST_DATA from './__testdata__/PokemonCard.json';
import { PokemonFragment } from '../src/queries/types/PokemonFragment';

describe('Pokemon Card', () => {
  test.each(TEST_DATA)('renders correctly for case %#', (pokemon) => {
    const { name, image } = pokemon;
    render(<PokemonCard pokemon={pokemon as PokemonFragment} />);
    const root = screen.getByTestId(`pokemon-card-${name}`);
    const nameInNode = root.querySelector('[data-testid=name]')?.innerHTML;
    const anchorNode = root.querySelector('a')?.getAttribute('href');
    const imageNode = root.querySelector('[data-testid=image]')?.getAttribute('src');
    expect(root).toMatchSnapshot();
    expect(nameInNode).toEqual(name);
    expect(anchorNode).toEqual(`/${name}`);
    expect(imageNode).toEqual(image);
  });
});
