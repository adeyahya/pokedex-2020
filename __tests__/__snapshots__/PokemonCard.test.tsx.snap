// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Card renders correctly for case 0 1`] = `
<div
  class="shadow hover:shadow-lg bg-white rounded-lg cursor-pointer"
  data-testid="pokemon-card-Bulbasaur"
>
  <figure
    class="h-40 md:h-32 lg:h-40 flex items-center justify-center pt-4"
  >
    <img
      alt="Bulbasaur"
      class="h-full"
      data-testid="image"
      src="https://img.pokemondb.net/artwork/bulbasaur.jpg"
    />
  </figure>
  <div
    class="p-4"
  >
    <a
      href="/pokemon/Bulbasaur"
    >
      <h3
        class="text-lg font-bold mb-0"
        data-testid="name"
      >
        Bulbasaur
         
        <span
          class="text-gray-500"
        >
          #
          001
        </span>
      </h3>
    </a>
    <span
      class="text-sm italic mb-2 block"
      data-testid="classification"
    >
      Seed Pokémon
    </span>
    <div
      class="mb-3"
    >
      <div
        data-testid="spec-list-types"
      >
        <h4
          class="text-sm font-bold mb-2"
          data-testid="spec-list-label"
        >
          types
        </h4>
        <div
          class="flex flex-row"
        >
          <span
            class="mr-2 bg-teal-200 px-2 text-sm font-medium rounded-full"
            data-testid="spec-list-value"
          >
            Grass
          </span>
          <span
            class="mr-2 bg-teal-200 px-2 text-sm font-medium rounded-full"
            data-testid="spec-list-value"
          >
            Poison
          </span>
        </div>
      </div>
    </div>
    <div
      class="mt-2"
    >
      <div
        class="flex justify-start flex-wrap"
        data-testid="specs"
      >
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MinWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              6.04kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              7.76kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxHP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              1071
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxCP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              951
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pokemon Card renders correctly for case 1 1`] = `
<div
  class="shadow hover:shadow-lg bg-white rounded-lg cursor-pointer"
  data-testid="pokemon-card-Ivysaur"
>
  <figure
    class="h-40 md:h-32 lg:h-40 flex items-center justify-center pt-4"
  >
    <img
      alt="Ivysaur"
      class="h-full"
      data-testid="image"
      src="https://img.pokemondb.net/artwork/ivysaur.jpg"
    />
  </figure>
  <div
    class="p-4"
  >
    <a
      href="/pokemon/Ivysaur"
    >
      <h3
        class="text-lg font-bold mb-0"
        data-testid="name"
      >
        Ivysaur
         
        <span
          class="text-gray-500"
        >
          #
          002
        </span>
      </h3>
    </a>
    <span
      class="text-sm italic mb-2 block"
      data-testid="classification"
    >
      Seed Pokémon
    </span>
    <div
      class="mb-3"
    >
      <div
        data-testid="spec-list-types"
      >
        <h4
          class="text-sm font-bold mb-2"
          data-testid="spec-list-label"
        >
          types
        </h4>
        <div
          class="flex flex-row"
        >
          <span
            class="mr-2 bg-teal-200 px-2 text-sm font-medium rounded-full"
            data-testid="spec-list-value"
          >
            Grass
          </span>
          <span
            class="mr-2 bg-teal-200 px-2 text-sm font-medium rounded-full"
            data-testid="spec-list-value"
          >
            Poison
          </span>
        </div>
      </div>
    </div>
    <div
      class="mt-2"
    >
      <div
        class="flex justify-start flex-wrap"
        data-testid="specs"
      >
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MinWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              11.38kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              14.63kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxHP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              1632
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxCP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              1483
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pokemon Card renders correctly for case 2 1`] = `
<div
  class="shadow hover:shadow-lg bg-white rounded-lg cursor-pointer"
  data-testid="pokemon-card-Venusaur"
>
  <figure
    class="h-40 md:h-32 lg:h-40 flex items-center justify-center pt-4"
  >
    <img
      alt="Venusaur"
      class="h-full"
      data-testid="image"
      src="https://img.pokemondb.net/artwork/venusaur.jpg"
    />
  </figure>
  <div
    class="p-4"
  >
    <a
      href="/pokemon/Venusaur"
    >
      <h3
        class="text-lg font-bold mb-0"
        data-testid="name"
      >
        Venusaur
         
        <span
          class="text-gray-500"
        >
          #
          003
        </span>
      </h3>
    </a>
    <span
      class="text-sm italic mb-2 block"
      data-testid="classification"
    >
      Seed Pokémon
    </span>
    <div
      class="mb-3"
    >
      <div
        data-testid="spec-list-types"
      >
        <h4
          class="text-sm font-bold mb-2"
          data-testid="spec-list-label"
        >
          types
        </h4>
        <div
          class="flex flex-row"
        >
          <span
            class="mr-2 bg-teal-200 px-2 text-sm font-medium rounded-full"
            data-testid="spec-list-value"
          >
            Grass
          </span>
          <span
            class="mr-2 bg-teal-200 px-2 text-sm font-medium rounded-full"
            data-testid="spec-list-value"
          >
            Poison
          </span>
        </div>
      </div>
    </div>
    <div
      class="mt-2"
    >
      <div
        class="flex justify-start flex-wrap"
        data-testid="specs"
      >
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MinWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              87.5kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              112.5kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxHP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              2580
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxCP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              2392
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pokemon Card renders correctly for case 3 1`] = `
<div
  class="shadow hover:shadow-lg bg-white rounded-lg cursor-pointer"
  data-testid="pokemon-card-Charmander"
>
  <figure
    class="h-40 md:h-32 lg:h-40 flex items-center justify-center pt-4"
  >
    <img
      alt="Charmander"
      class="h-full"
      data-testid="image"
      src="https://img.pokemondb.net/artwork/charmander.jpg"
    />
  </figure>
  <div
    class="p-4"
  >
    <a
      href="/pokemon/Charmander"
    >
      <h3
        class="text-lg font-bold mb-0"
        data-testid="name"
      >
        Charmander
         
        <span
          class="text-gray-500"
        >
          #
          004
        </span>
      </h3>
    </a>
    <span
      class="text-sm italic mb-2 block"
      data-testid="classification"
    >
      Lizard Pokémon
    </span>
    <div
      class="mb-3"
    >
      <div
        data-testid="spec-list-types"
      >
        <h4
          class="text-sm font-bold mb-2"
          data-testid="spec-list-label"
        >
          types
        </h4>
        <div
          class="flex flex-row"
        >
          <span
            class="mr-2 bg-teal-200 px-2 text-sm font-medium rounded-full"
            data-testid="spec-list-value"
          >
            Fire
          </span>
        </div>
      </div>
    </div>
    <div
      class="mt-2"
    >
      <div
        class="flex justify-start flex-wrap"
        data-testid="specs"
      >
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MinWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              7.44kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              9.56kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxHP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              955
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxCP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              841
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pokemon Card renders correctly for case 4 1`] = `
<div
  class="shadow hover:shadow-lg bg-white rounded-lg cursor-pointer"
  data-testid="pokemon-card-Charmeleon"
>
  <figure
    class="h-40 md:h-32 lg:h-40 flex items-center justify-center pt-4"
  >
    <img
      alt="Charmeleon"
      class="h-full"
      data-testid="image"
      src="https://img.pokemondb.net/artwork/charmeleon.jpg"
    />
  </figure>
  <div
    class="p-4"
  >
    <a
      href="/pokemon/Charmeleon"
    >
      <h3
        class="text-lg font-bold mb-0"
        data-testid="name"
      >
        Charmeleon
         
        <span
          class="text-gray-500"
        >
          #
          005
        </span>
      </h3>
    </a>
    <span
      class="text-sm italic mb-2 block"
      data-testid="classification"
    >
      Flame Pokémon
    </span>
    <div
      class="mb-3"
    >
      <div
        data-testid="spec-list-types"
      >
        <h4
          class="text-sm font-bold mb-2"
          data-testid="spec-list-label"
        >
          types
        </h4>
        <div
          class="flex flex-row"
        >
          <span
            class="mr-2 bg-teal-200 px-2 text-sm font-medium rounded-full"
            data-testid="spec-list-value"
          >
            Fire
          </span>
        </div>
      </div>
    </div>
    <div
      class="mt-2"
    >
      <div
        class="flex justify-start flex-wrap"
        data-testid="specs"
      >
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MinWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              16.63kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              21.38kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxHP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              1557
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxCP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              1411
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pokemon Card renders correctly for case 5 1`] = `
<div
  class="shadow hover:shadow-lg bg-white rounded-lg cursor-pointer"
  data-testid="pokemon-card-Charizard"
>
  <figure
    class="h-40 md:h-32 lg:h-40 flex items-center justify-center pt-4"
  >
    <img
      alt="Charizard"
      class="h-full"
      data-testid="image"
      src="https://img.pokemondb.net/artwork/charizard.jpg"
    />
  </figure>
  <div
    class="p-4"
  >
    <a
      href="/pokemon/Charizard"
    >
      <h3
        class="text-lg font-bold mb-0"
        data-testid="name"
      >
        Charizard
         
        <span
          class="text-gray-500"
        >
          #
          006
        </span>
      </h3>
    </a>
    <span
      class="text-sm italic mb-2 block"
      data-testid="classification"
    >
      Flame Pokémon
    </span>
    <div
      class="mb-3"
    >
      <div
        data-testid="spec-list-types"
      >
        <h4
          class="text-sm font-bold mb-2"
          data-testid="spec-list-label"
        >
          types
        </h4>
        <div
          class="flex flex-row"
        >
          <span
            class="mr-2 bg-teal-200 px-2 text-sm font-medium rounded-full"
            data-testid="spec-list-value"
          >
            Fire
          </span>
          <span
            class="mr-2 bg-teal-200 px-2 text-sm font-medium rounded-full"
            data-testid="spec-list-value"
          >
            Flying
          </span>
        </div>
      </div>
    </div>
    <div
      class="mt-2"
    >
      <div
        class="flex justify-start flex-wrap"
        data-testid="specs"
      >
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MinWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              79.19kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              101.81kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxHP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              2602
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxCP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              2413
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pokemon Card renders correctly for case 6 1`] = `
<div
  class="shadow hover:shadow-lg bg-white rounded-lg cursor-pointer"
  data-testid="pokemon-card-Squirtle"
>
  <figure
    class="h-40 md:h-32 lg:h-40 flex items-center justify-center pt-4"
  >
    <img
      alt="Squirtle"
      class="h-full"
      data-testid="image"
      src="https://img.pokemondb.net/artwork/squirtle.jpg"
    />
  </figure>
  <div
    class="p-4"
  >
    <a
      href="/pokemon/Squirtle"
    >
      <h3
        class="text-lg font-bold mb-0"
        data-testid="name"
      >
        Squirtle
         
        <span
          class="text-gray-500"
        >
          #
          007
        </span>
      </h3>
    </a>
    <span
      class="text-sm italic mb-2 block"
      data-testid="classification"
    >
      Tiny Turtle Pokémon
    </span>
    <div
      class="mb-3"
    >
      <div
        data-testid="spec-list-types"
      >
        <h4
          class="text-sm font-bold mb-2"
          data-testid="spec-list-label"
        >
          types
        </h4>
        <div
          class="flex flex-row"
        >
          <span
            class="mr-2 bg-teal-200 px-2 text-sm font-medium rounded-full"
            data-testid="spec-list-value"
          >
            Water
          </span>
        </div>
      </div>
    </div>
    <div
      class="mt-2"
    >
      <div
        class="flex justify-start flex-wrap"
        data-testid="specs"
      >
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MinWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              7.88kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              10.13kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxHP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              1008
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxCP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              891
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pokemon Card renders correctly for case 7 1`] = `
<div
  class="shadow hover:shadow-lg bg-white rounded-lg cursor-pointer"
  data-testid="pokemon-card-Wartortle"
>
  <figure
    class="h-40 md:h-32 lg:h-40 flex items-center justify-center pt-4"
  >
    <img
      alt="Wartortle"
      class="h-full"
      data-testid="image"
      src="https://img.pokemondb.net/artwork/wartortle.jpg"
    />
  </figure>
  <div
    class="p-4"
  >
    <a
      href="/pokemon/Wartortle"
    >
      <h3
        class="text-lg font-bold mb-0"
        data-testid="name"
      >
        Wartortle
         
        <span
          class="text-gray-500"
        >
          #
          008
        </span>
      </h3>
    </a>
    <span
      class="text-sm italic mb-2 block"
      data-testid="classification"
    >
      Turtle Pokémon
    </span>
    <div
      class="mb-3"
    >
      <div
        data-testid="spec-list-types"
      >
        <h4
          class="text-sm font-bold mb-2"
          data-testid="spec-list-label"
        >
          types
        </h4>
        <div
          class="flex flex-row"
        >
          <span
            class="mr-2 bg-teal-200 px-2 text-sm font-medium rounded-full"
            data-testid="spec-list-value"
          >
            Water
          </span>
        </div>
      </div>
    </div>
    <div
      class="mt-2"
    >
      <div
        class="flex justify-start flex-wrap"
        data-testid="specs"
      >
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MinWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              19.69kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              25.31kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxHP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              1582
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxCP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              1435
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pokemon Card renders correctly for case 8 1`] = `
<div
  class="shadow hover:shadow-lg bg-white rounded-lg cursor-pointer"
  data-testid="pokemon-card-Blastoise"
>
  <figure
    class="h-40 md:h-32 lg:h-40 flex items-center justify-center pt-4"
  >
    <img
      alt="Blastoise"
      class="h-full"
      data-testid="image"
      src="https://img.pokemondb.net/artwork/blastoise.jpg"
    />
  </figure>
  <div
    class="p-4"
  >
    <a
      href="/pokemon/Blastoise"
    >
      <h3
        class="text-lg font-bold mb-0"
        data-testid="name"
      >
        Blastoise
         
        <span
          class="text-gray-500"
        >
          #
          009
        </span>
      </h3>
    </a>
    <span
      class="text-sm italic mb-2 block"
      data-testid="classification"
    >
      Shellfish Pokémon
    </span>
    <div
      class="mb-3"
    >
      <div
        data-testid="spec-list-types"
      >
        <h4
          class="text-sm font-bold mb-2"
          data-testid="spec-list-label"
        >
          types
        </h4>
        <div
          class="flex flex-row"
        >
          <span
            class="mr-2 bg-teal-200 px-2 text-sm font-medium rounded-full"
            data-testid="spec-list-value"
          >
            Water
          </span>
        </div>
      </div>
    </div>
    <div
      class="mt-2"
    >
      <div
        class="flex justify-start flex-wrap"
        data-testid="specs"
      >
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MinWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              74.81kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              96.19kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxHP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              2542
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxCP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              2355
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pokemon Card renders correctly for case 9 1`] = `
<div
  class="shadow hover:shadow-lg bg-white rounded-lg cursor-pointer"
  data-testid="pokemon-card-Caterpie"
>
  <figure
    class="h-40 md:h-32 lg:h-40 flex items-center justify-center pt-4"
  >
    <img
      alt="Caterpie"
      class="h-full"
      data-testid="image"
      src="https://img.pokemondb.net/artwork/caterpie.jpg"
    />
  </figure>
  <div
    class="p-4"
  >
    <a
      href="/pokemon/Caterpie"
    >
      <h3
        class="text-lg font-bold mb-0"
        data-testid="name"
      >
        Caterpie
         
        <span
          class="text-gray-500"
        >
          #
          010
        </span>
      </h3>
    </a>
    <span
      class="text-sm italic mb-2 block"
      data-testid="classification"
    >
      Worm Pokémon
    </span>
    <div
      class="mb-3"
    >
      <div
        data-testid="spec-list-types"
      >
        <h4
          class="text-sm font-bold mb-2"
          data-testid="spec-list-label"
        >
          types
        </h4>
        <div
          class="flex flex-row"
        >
          <span
            class="mr-2 bg-teal-200 px-2 text-sm font-medium rounded-full"
            data-testid="spec-list-value"
          >
            Bug
          </span>
        </div>
      </div>
    </div>
    <div
      class="mt-2"
    >
      <div
        class="flex justify-start flex-wrap"
        data-testid="specs"
      >
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MinWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              2.54kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxWg
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              3.26kg
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxHP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              443
            </span>
          </div>
        </div>
        <div
          class="flex flex-initial flex-col mr-2 mt-2"
        >
          <span
            class="text-xs font-bold"
            data-testid="spec-name"
          >
            MaxCP
          </span>
          <div>
            <span
              class="bg-green-400 p-1 font-bold rounded-lg text-xs"
              data-testid="spec-value"
            >
              367
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
